### PR TITLE
[WIP] Trying to fix block structure mismatch

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

**Comment by alesapin** [comment]
Corresponding issue https://github.com/yandex/ClickHouse/issues/3162

**Comment by KochetovNicolai** [comment]
00747_contributors

File: dbms/src/Interpreters/InterpreterSelectQuery.cpp [hunk 1]
```diff
@@ -1236,6 +1236,17 @@ void InterpreterSelectQuery::executeMergeSorted(Pipeline & pipeline)
     /// If there are several streams, then we merge them into one
     if (pipeline.hasMoreThanOneStream())
     {
+        /// Unify streams in case they have different headers.
+        auto first_header = pipeline.streams.at(0)->getHeader();
```
Comment (alesapin) [inline dbms/src/Interpreters/InterpreterSelectQuery.cpp]: Maybe unify this code with same for Union stream?

Comment (alexey-milovidov) [reply dbms/src/Interpreters/InterpreterSelectQuery.cpp]: Yes, you can add unification in Union stream.

**Pull request merged**